src/hotspot/share/classfile/stringTable.cpp

```diff
@@ -801,11 +801,11 @@ oop StringTable::lookup_shared(const jchar* name, int len) {
 // to guarantee because CDS runs with a single Java thread. See JDK-8253495.)
 void StringTable::allocate_shared_strings_array(TRAPS) {
   assert(DumpSharedSpaces, "must be");
-  if (_items_count > (size_t)max_jint) {
-    fatal("Too many strings to be archived: " SIZE_FORMAT, _items_count);
+  if (_items_count > max_jint) {
```

Comment:
This is impossible given `_items_count_` is now an int

Reply:
Maybe we still want items_count to be a size_t then so we can detect this.